fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindDynamicSplit.java

```diff
@@ -0,0 +1,196 @@
+import org.apache.doris.nereids.trees.expressions.GreaterThanEqual;
+import org.apache.doris.nereids.trees.expressions.LessThanEqual;
+import org.apache.doris.nereids.trees.expressions.Slot;
+import org.apache.doris.nereids.trees.expressions.SlotReference;
+import org.apache.doris.nereids.trees.expressions.literal.Literal;
+import org.apache.doris.nereids.trees.plans.Plan;
+import org.apache.doris.nereids.trees.plans.commands.info.SplitColumnInfo;
+import org.apache.doris.nereids.trees.plans.logical.LogicalCatalogRelation;
+import org.apache.doris.nereids.trees.plans.logical.LogicalDynamicSplit;
+import org.apache.doris.nereids.trees.plans.logical.LogicalFilter;
+import org.apache.doris.nereids.trees.plans.logical.LogicalProject;
+import org.apache.doris.nereids.trees.plans.logical.LogicalRelation;
+import org.apache.doris.nereids.trees.plans.logical.LogicalSubQueryAlias;
+import org.apache.doris.nereids.trees.plans.logical.LogicalView;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import org.apache.commons.lang3.Range;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+/**
+ * This class defines the rule to bind a dynamic split operation in the query plan.
+ */
+public class BindDynamicSplit implements AnalysisRuleFactory {
+    private static final Logger LOG = LoggerFactory.getLogger(BindDynamicSplit.class);
+
+    @Override
+    public List<Rule> buildRules() {
+        return ImmutableList.of(
+                //LogicalDynamicSplit -> LogicalFilter -> LogicalCatalogRelation
+                RuleType.BIND_DYNAMIC_SPLIT.build(logicalDynamicSplit(logicalFilter(logicalRelation()))
+                        .thenApply(ctx -> {
+
+                            Plan result = ctx.root.child();
+                            LogicalDynamicSplit<LogicalFilter<LogicalRelation>> logicalDynamicSplit = ctx.root;
+                            LogicalRelation logicalRelation = (LogicalRelation) ((LogicalFilter<?>)
+                                    logicalDynamicSplit.child())
+                                    .child();
+                            return rewritePlan(logicalRelation, result, logicalDynamicSplit);
+                        })),
+                //LogicalDynamicSplit -> LogicalProject -> LogicalFilter -> LogicalCatalogRelation
+                RuleType.BIND_DYNAMIC_SPLIT.build(logicalDynamicSplit(logicalProject(logicalFilter(logicalRelation())))
+                        .thenApply(ctx -> {
+
+                            Plan result = ctx.root.child();
+                            LogicalDynamicSplit logicalDynamicSplit = ctx.root;
+                            LogicalRelation logicalRelation = (LogicalRelation) ((LogicalProject<?>)
+                                    ((LogicalFilter<?>) logicalDynamicSplit.child()).child()).child();
+                            return rewritePlan(logicalRelation, result, logicalDynamicSplit);
```

Comment:
you can not direct process the logicalRelation because the logical project maybe change the column name of the relation?

say
```sql
select id as my_id, from tbl
```
and we want to split by my_id, you can not find my_id in the tbl

Reply:
When creating a batch job, the split column must be specified, and it must be the actual column name. Therefore, there will be no cases where an alias is used for the split column.